plugin/src/init.server.lua

```diff
@@ -2,24 +2,65 @@
 local HttpService = game:GetService("HttpService")
 assert(plugin, "This code must run inside of a plugin")
 
+if game:GetService("RunService"):IsRunning() then
+	return
+end
+
 local toolbar = plugin:CreateToolbar("Luau") :: PluginToolbar
-local button = toolbar:CreateButton(
+
+local ConnectButton = toolbar:CreateButton(
 	"Luau Language Server Setup",
 	"Toggle Menu",
 	"rbxassetid://11115506617",
 	"Luau Language Server"
 ) :: PluginToolbarButton
 
-local widgetInfo = DockWidgetPluginGuiInfo.new(
-	-- widget info
-	Enum.InitialDockState.Float,
-	false,
-	false,
-	300,
-	200,
-	120,
-	70
-)
+local SettingsButton =
+	toolbar:CreateButton("Settings", "Open Settings", "rbxassetid://13997395868") :: PluginToolbarButton
+
+--#region Settings
+
+local AnalyticsService = game:GetService("AnalyticsService")
+
+local Settings = nil :: any
+
+local SettingsModule = AnalyticsService:FindFirstChild("LuauLSP_Settings")
+if not SettingsModule then
+	SettingsModule = script.DefaultSettings:Clone()
+	SettingsModule.Name = "LuauLSP_Settings"
+	SettingsModule.Parent = AnalyticsService
+end
+
+local function LoadSettings()
+	local result, parseError: any = loadstring(SettingsModule.Source)
+	if result == nil then
+		warn("[Luau Language Server] Could not load settings: " .. parseError)
+		Settings = nil
+		return
+	end
+	pcall(function()
+		Settings = result()
+	end)
```

Comment:
Is there any error that can occur here which should be appropriately shown to the user?

Reply:
I didn't encountered anything here when testing and I also handle the settings error on watchChanges before connecting

Reply:
Is it necessary for this to be wrapped in a pcall then, if no error occurs?

Reply:
did some tests, we already handle errors on 37 but, that pcall solves a really edge case like if person put `error()` on the setting and reload the session they can't open the setting menu again, so I think we can keep the pcall

Reply:
If we intend to keep it, we should probably re-emit the error as a warning, so that it isn't silently caught and the user knows why it's not working